WireAuthentication/Sources/WireAuthenticationUI/Views/Login/LoginViaSSO/LoginViaSSOViewModel.swift

```diff
@@ -22,4 +22,107 @@ import SwiftUI
 import WireAuthenticationAPI
 
 @MainActor
-public final class LoginViaSSOViewModel: ObservableObject {}
+package final class LoginViaSSOViewModel: ObservableObject {
+
+    /// Generate the link to the SSO authentication screen
+    ///
+    /// - Parameters:
+    ///   - baseURL: Backend URL.
+    ///   - ssoCode: SSO code
+    ///   - callbackScheme: The URL scheme that where the callback will be provided.
+    /// - Returns: URL to the SSO authentication screen
+    func buildSSOLink(baseURL: URL, ssoCode: UUID, callbackScheme: String) async throws -> URL {
```

Comment:
question: how do you envision this to be called? When the view appears? And where would the `ssoCode` and `callbackScheme` come from? The view model itself?

I'm wondering if it would be good to put this inside something like `SSOLinkGenerator` that can be invoked from here.